### DISCUSSION TITLE
What is caffeine dependence on Kotlin?

### DISCUSSION THREAD

**Discussion by jflebeau** [ben-manes/caffeine#858] [Q&A] [answered]
I didn't look too deep into it, but I didn't see any explicit mention of it, trying to avoid Kotlin runtime version conflicts...

**Comment by ben-manes** [comment, marked as answer]
Sorry, this was a metadata leak from the build system and not a compile / runtime dependency. I was receiving CVE alerts for test and build plugins that used older versions of Kotlin, so I used a dependency constraint to force the minimum version. This way the build would be less as susceptible to supply chain attacks (e.g. codecov, circleci, solarwinds exploits). In Maven this is simply the an addition to the dependencyManagement section that does not impact resolution outside of that dependency itself, so noise. In Gradle they now prefer their own module metadata over the maven pom where the constraint gets applied globally to the project.

Anyway, the build was fixed to no longer leak this in future releases. If it causes a lot of headache then I can cut a new release, else you can simply ignore it in your build. The library does not have any required dependencies, just static analysis annotations that are optional.

Fix commits: https://github.com/ben-manes/caffeine/commit/396a9405f52d8f7bd3baa162fb6f1a9f5a239c91, https://github.com/ben-manes/caffeine/commit/7321a56a0fe92308c5b20c2a789ab77907ca480e
